webapp/src/ts/modules/reports/reports-content.component.html

```diff
@@ -34,107 +34,110 @@
           <div>{{summaries[$index] | title:forms}}</div>
           <div class="position small-font"
-               *ngIf="summaries[$index].subject.lineage || selection.lineage || selection.contact.parent || selection.place || (!selection.form && selection.name)"
-               ng-bind-html="summaries[$index].subject.lineage || selection.lineage || selection.contact.parent || selection.place | lineage">
+               *ngIf="summaries[$index].subject?.lineage || selection.lineage || selection.contact?.parent || selection.place || (!selection.form && selection.name)"
+               [innerHTML]="summaries[$index].subject?.lineage || selection.lineage || selection.contact?.parent || selection.place | lineage">
           </div>
-          <!--<mm-sender *ngIf="summaries[$index].subject.type" message="selection" hide-lineage="true" sent-by="true"></mm-sender>-->
-          <div class="time small-font" ng-bind-html="summaries[$index].reported_date | relativeDate"></div>
+          <mm-sender *ngIf="summaries[$index].subject?.type" [message]="selection" [hideLineage]="true" [sentBy]="true"></mm-sender>
+          <div class="time small-font" [innerHTML]="summaries[$index].reported_date | relativeDate"></div>
         </div>
       </div>
 
-      <div ng-show="selection.loading">
+      <div *ngIf="selection.loading">
         <div class="loader"></div>
       </div>
 
-      <!--<div class="details" *ngIf="selection.expanded">
+      <div class="details" *ngIf="selection.expanded">
         <ul>
-          <li *ngIf="selection.formatted.form" ng-repeat="field in selection.formatted.content_type === 'xml' ? selection.formatted.fields : selection.formatted.fields.data" class="indent-{{field.depth || 0}}">
-            <label>
-              <span *ngIf="selection.formatted.content_type === 'xml'" translate>{{field.label}}</span>
-              <span *ngIf="selection.formatted.content_type !== 'xml'">{{field.label}}</span>
-              <span class="fa fa-cogs" *ngIf="field.generated" translate translate-attr-title="Generated report field"></span>
-            </label>
-            <p *ngIf="!field.imagePath">
-              <span *ngIf="field.target && field.target.filter"><a ng-click="search(field.target.filter)">{{field.value}}</a></span>
-              <span *ngIf="field.target && field.target.url"><a ui-sref="{{field.target.url.route}}({{field.target.url.params}})">{{field.value}}</a></span>
-              <span *ngIf="!field.target">{{field.value}}</span>
-            </p>
-            <report-image *ngIf="field.imagePath" report="{{selection.formatted._id}}" path="{{field.imagePath}}"/>
-          </li>
-          <li *ngIf="selection.formatted.errors.length">
+          <ng-container *ngIf="selection.formatted?.form">
+            <li *ngFor="let field of (selection.formatted.content_type === 'xml' ? selection.formatted.fields : selection.formatted.fields.data)" class="indent-{{field.depth || 0}}">
```

Comment:
Is it possible to add trackBy? That improves a bit the performance of ngFor

Reply:
I'm not sure what I'd track it by, we care about every property on the field and we'll likely want to update the view with every change. Do you think it's worth to serialize all field keys in a track by?